share/lib/api_call.sh

```diff
@@ -30,3 +31,37 @@ simple_api_call() {
     exit 1
   fi
 }
+
+# retrieve one entry from ~/.rudder
+_get_conf() {
+  conf="$1"
+  name="$2"
+  conffile=~/.rudder
+  [ -f "${conffile}" ] || return
+  [ -z "${conf}" ] && conf=".*"
+  # extract inifile section | extract value
+  sed -n "/^\\[${conf}\\]$/,/^\\[.*\\]$/p" "${conffile}" | sed -n "/^${name} *= */s/^${name} *= *//p"
+}
+
+# This function calls the api with a token
+full_api_call() {
+  api="$1"
+  url="$2"
+  conf="$3"
+  token="$4"
+
+  if [ -z "${url}" ]
+  then
+    url=`_get_conf "${conf}" "url"`
+    if [ -z "${url}" ]
+    then
+      host=`cat /var/rudder/cfengine-community/policy_server.dat 2>/dev/null`
+      [ -z "${host}" ] && host="localhost"
+      url="https://${host}/rudder"
+    fi
+  fi
+
+  [ -z "${token}" ] && token=`_get_conf "${conf}" "token"`
```

Comment:
I think you should add an explicit check that the token isn't still empty here, and print an error message if so. Otherwise, the user will see (at best) an error message from the API itself, although it's easy to tell them they need a token in ~/.rudder or with the -t option at this stage.